lib/Dialect/Arc/Transforms/GroupResetsAndEnables.cpp

```diff
@@ -114,6 +116,76 @@ struct EnableGroupingPattern : public OpRewritePattern<ClockTreeOp> {
     return success(changed);
   }
 };
+
+/// Where possible without domination issues, group assignments inside IfOps and
+/// return true if any operations were moved.
+bool groupInRegion(Region *region, Operation *clockTreeOp,
+                   PatternRewriter *rewriter) {
+  bool changed = false;
+  if (region->empty())
+    return changed;
+
+  // We assume this function is given only IfOp then/else regions, so we have at
+  // most 1 block
+  Block &block = region->front();
```

Comment:
Nit: we could take a `Block*` as argument which would get rid of the assumption that this only operates on regions with a single block. You can then check `if (!block) return false` which allows you to use `ifOp.elseBlock()` at the use-site which will return a `nullptr` when the region is empty.

Reply:
Oh great point, thanks!